syft/format/cyclonedxjson/decoder.go

```diff
@@ -25,10 +26,12 @@ func NewFormatDecoder() sbom.FormatDecoder {
 	}
 }
 
-func (d decoder) Decode(reader io.ReadSeeker) (*sbom.SBOM, sbom.FormatID, string, error) {
-	if reader == nil {
-		return nil, "", "", fmt.Errorf("no SBOM bytes provided")
+func (d decoder) Decode(r io.Reader) (*sbom.SBOM, sbom.FormatID, string, error) {
+	reader, err := stream.SeekableReader(r)
```

Comment:
Should wrapping the reader in something so it can be reset happen at the `syft/format`.`Decode` function instead of per-format?